src/backend/out.c

```diff
@@ -159,7 +159,8 @@ void outdata(symbol *s)
                 //printf("DT_azeros %d\n", dt->DTazeros);
             case_azeros:
                 datasize += dt->DTazeros;
-                if (dt == dtstart && !dt->DTnext && s->Sclass != SCcomdat)
+                if (dt == dtstart && !dt->DTnext && s->Sclass != SCcomdat &&
+                    (s->Sseg == UNKNOWN || s->Sseg <= UDATA))
```

Comment:
This code puts DTazeros into the BSS segment. There is code in objc_glue.c that was assigning a different seg number, and expecting the data to be put into that segment. Expanding the azeros optimizations broke that. This fixes it.